File: JavaScript/Lv.0_이어 붙인 수.js
// JavaScript

function solution(num_list) {
    let oddNum = '', evenNum = '';
    
    num_list.map((value, i) => {
        if (value % 2 === 1) { // 홀수
            oddNum += String(value);
        } else if (value % 2 === 0) { // 짝수
            evenNum += String(value);
        }
    })
    
    console.log("oddNum : ",oddNum, "evenNum : ",evenNum);
    return Number(oddNum) + Number(evenNum);
}

// 아래와 같이 reduce로 누적하는 방안도 있음
function solution(num_list) {
    const { odds, evens } = num_list.reduce(({ odds, evens }, num) => {
        if (num % 2 === 0) {
            evens.push(num)
        } else {
            odds.push(num)
        }
        return { odds, evens }
    }, { odds: [], evens: [] })
    return Number(odds.join('')) + Number(evens.join(''))
}

File: Java/Lv.0_이어 붙인 수.java
// Java

class Solution {
    public int solution(int[] num_list) {
        int answer = 0;
        String oddNum = "";
        String evenNum = "";
        
        for (int i : num_list) {
            oddNum += i % 2 == 1 ? i+"" : "";
            evenNum += i % 2 == 0 ? i+"" : "";
        }
        
        return Integer.parseInt(oddNum) + Integer.parseInt(evenNum);
    }
}

// Int => String 변환 방식은 String.valueOf(i)와 Integer.toString(i)가 있다.